components/micro-gateway-core/src/main/ballerina/src/gateway/analytics/faults_util.bal

```diff
@@ -21,19 +21,30 @@ public function getFaultMetaData(FaultDTO dto) returns string {
     return dto.metaClientType;
 }
 
-public function getFaultPayloadData(FaultDTO dto) returns string {
-    return dto.consumerKey + OBJ + dto.apiName + OBJ + dto.apiVersion + OBJ + dto.apiContext + OBJ +
+public function getFaultPayloadData(FaultDTO dto, string amAnalyticsVersion) returns string {
+    printDebug(KEY_ANALYTICS_FILTER, "Generating fault data payload for " + amAnalyticsVersion);
+    string payloadData = dto.consumerKey + OBJ + dto.apiName + OBJ + dto.apiVersion + OBJ + dto.apiContext + OBJ +
     dto.resourcePath + OBJ + dto.method + OBJ + dto.apiCreator + OBJ + dto.userName + OBJ + dto.userTenantDomain + OBJ +
     dto.apiCreatorTenantDomain + OBJ + dto.hostName + OBJ + dto.applicationId + OBJ +
-    dto.applicationName + OBJ + dto.protocol + OBJ + dto.errorCode.toString() + OBJ + dto.errorMessage + OBJ + dto.faultTime.toString();
+    dto.applicationName + OBJ + dto.protocol + OBJ + dto.errorCode.toString() + OBJ + dto.errorMessage + OBJ +
+    dto.faultTime.toString();
+
+    if (amAnalyticsVersion == "3.1.0") {
```

Comment:
Lets not make this exact match. Upto 3.1.0 should be the condition

Reply:
Fixed with https://github.com/wso2/product-microgateway/pull/1319/commits/83140f71d678b0f9f98a606c1f991997d0c0bd82